packages/block-editor/src/components/block-context/README.md

```diff
@@ -0,0 +1,57 @@
+Block Context
+=============
+
+Block Context is a React implementation of WordPress's block context. Block context, much like [React's context](https://reactjs.org/docs/context.html), is a method for passing and inheriting values deeply through a hierarchy of blocks. Because of the similarities with React's context, the client-side implementation here is quite minimal. It is complemented by equivalent behaviors in the server-side rendering of a block.
+
+Note that the implementation of Block Context is distinct from [the `BlockEdit` context](../block-edit). While it is true that both provide context relevant for the editing of a block, Block Context is implemented separately so as to prioritize it as most identifiable amongst the machinery of block context, and not amongst other client-side editing context of a block.
```

Comment:
The filter exposed in the same component is `'editor.BlockListBlock'`, so I echo your proposal.

Reply:
> This paragraph inspired me to grep for `BlockContext` which turned up this:

This wasn't even the context I was referring to in the text here 😅 Maybe it's what you intended to highlight. The one I was referring to in the text was this one:

https://github.com/WordPress/gutenberg/blob/516f0226c150259eb682fde1eb9a287e6ce9d9ee/packages/block-editor/src/components/block-edit/context.js

I've not yet digested if and how this one differs, I'd agree that a rename could help clarify it as specific to the block list. I can create a tracking issue while I'm writing the others 😛

Reply:
we actually have a third one called BlockEditContext (or something like that). I think we need to be very careful about the naming and whether we could merge some of these together.

Reply:
> we actually have a third one called BlockEditContext (or something like that). I think we need to be very careful about the naming and whether we could merge some of these together.

See also https://github.com/WordPress/gutenberg/pull/21467#discussion_r406773910

Reply:
> This paragraph inspired me to grep for `BlockContext` which turned up this:
> 
> https://github.com/WordPress/gutenberg/blob/516f0226c150259eb682fde1eb9a287e6ce9d9ee/packages/block-editor/src/components/block-list/block.js#L34
> 
> It might be worth renaming this to `BlockListBlockContext` (or something) so that we don't get confused.

Tracking: #21664